clustering/common/src/main/java/org/jboss/as/clustering/controller/ReadAttributeTranslationHandler.java

```diff
@@ -48,13 +49,24 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
         PathAddress targetAddress = this.translation.getPathAddressTransformation().apply(currentAddress);
         Attribute targetAttribute = this.translation.getTargetAttribute();
         ModelNode targetOperation = Operations.createReadAttributeOperation(targetAddress, targetAttribute);
-        ImmutableManagementResourceRegistration targetRegistration = this.translation.getResourceRegistrationTransformation().apply(context.getResourceRegistration());
-        OperationStepHandler readAttributeHandler = targetRegistration.getAttributeAccess(PathAddress.EMPTY_ADDRESS, targetAttribute.getName()).getReadHandler();
+        ImmutableManagementResourceRegistration registration = (currentAddress == targetAddress) ? context.getResourceRegistration() : context.getRootResourceRegistration().getSubModel(targetAddress);
+        if (registration == null) {
+            throw new OperationFailedException(ControllerLogger.MGMT_OP_LOGGER.noSuchResourceType(targetAddress));
+        }
+        OperationStepHandler readAttributeHandler = registration.getAttributeAccess(PathAddress.EMPTY_ADDRESS, targetAttribute.getName()).getReadHandler();
         OperationStepHandler readTranslatedAttributeHandler = new ReadTranslatedAttributeStepHandler(readAttributeHandler, targetAttribute, this.translation.getReadTranslator());
         // If targetOperation applies to the current resource, we can execute in the current step
         if (targetAddress == currentAddress) {
             readTranslatedAttributeHandler.execute(context, targetOperation);
         } else {
+            if (registration.isRuntimeOnly()) {
+                try {
+                    context.readResourceFromRoot(targetAddress, false);
+                } catch (Resource.NoSuchResourceException ignore) {
+                    // If the target runtime resource does not exist return UNDEFINED
```

Comment:
@pferraro Why?

Reply:
This is a generic operation handler, used for deprecated attributes/metrics whose return value comes from a different (i.e. target) attribute, potentially at a different path address.  If the target attribute resides at a different path address, and the resource at the target address is a runtime resource, we need to handle the case where no resource exists at the target address, otherwise the subsequent OperationContext.addStep(...) would fail to execute.
If you consider the previous behavior of a given deprecated metric, the "no resource exists" condition is equivalent to the case where the target capability's service (from which the metric is obtained) is not started.  In this case, we would normally return a value of UNDEFINED.  When the target resource is a runtime resource, that runtime resource only exists when the target capability's service (from which the metric is obtained) is started - thus the result of the read-attribute translation is consistent with the previous behavior.

Reply:
OK. The previous behavior sounds a bit off but I think it's the way things are handled in many places (not just clustering) so changing that is out of scope for this PR.